### PR TITLE
Add token filter and action

### PR DESCRIPTION
**Pull request by gounux**
- add a token filter based on client type (e.g. `QField`, `QFieldSync`, `Browser`...)
- add a custom action to disable a token by setting it's expiration date to now

### CODE REVIEW

File: docker-app/qfieldcloud/authentication/admin.py [hunk 1]
```diff
@@ -1,9 +1,49 @@
...
+    parameter_name = "client_type"
+
+    def lookups(self, request, model_admin):
+        """
+        Returns a list of tuples. The first element in each
+        tuple is the coded value for the option that will
+        appear in the URL query. The second element is the
+        human-readable name for the option that will appear
+        in the right sidebar.
+        Here it is just the several available AuthToken.ClientType
```
Comment (suricactus) [inline docker-app/qfieldcloud/authentication/admin.py:23]:
```suggestion
        Here it is just the several available `AuthToken.ClientType`.
```

File: docker-app/qfieldcloud/authentication/admin.py [hunk 2]
```diff
@@ -15,6 +55,22 @@ class AuthTokenAdmin(admin.ModelAdmin):
...
+
+    actions = ("expire_selected_tokens",)
 
     search_fields = ("user__username__iexact", "client_type", "key__startswith")
+
+    def expire_selected_tokens(self, request, queryset):
+        """
+        Sets a set of tokens to expired
+        by updating the expires_at date to now
+        Expires only valid tokens
```
Comment (suricactus) [inline docker-app/qfieldcloud/authentication/admin.py:73]:
```suggestion
        Sets a set of tokens to expired by updating the `expires_at` date to now.
        Expires only valid tokens.
```

**Review by suricactus — APPROVED** [review]
Perfect! I had to do some nitpicking by commenting on the comments...

**Comment by m-kuhn** [comment]
Thanks for this! 🎉
Would it be possible to add type hints?

**Review by suricactus — APPROVED** [review]
Lovely!

**Pull request merged**